### PR TITLE
Pre conditions

### PR DESCRIPTION
**Pull request by jverbraeken**
- Relevant Issues: -
- Related Pull Requests: -

* * *

## What
This Pull Request adds to the repository pre-condition validation of all parameters of the project.

- Apache Commons is used to validate all input to public / protected / package-private methods
- asserts are used to validate the input to private methods

## Why
This Pull Request is needed because almost no parameters were validated before

## How
This feature can be viewed/tested within the project by verifying that everything still works

## Alternative implementation
Other implementations that I've have considered are

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java [hunk 1]
```diff
@@ -303,7 +294,29 @@ private BluetoothDevice getDevice(final String address) {
             }
             return null;
         }
-
     }
 
+    private final class CustomBroadcastReceiver extends BroadcastReceiver {
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java]: A more specific name would be appreciated

File: app/src/main/java/com/nervousfish/nervousfish/activities/RhythmCreateActivity.java [hunk 2]
```diff
@@ -247,6 +256,8 @@ private KMeansClusterHelper() {
          * @return The unique key that corresponds to the taps
          */
         int getEncryptionKey(final List<SingleTap> taps) {
+            Validate.notNull(taps);
+            Validate.isTrue(taps.size() > 2);
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/activities/RhythmCreateActivity.java]: This should be 2 >= 1

File: fooaccountInformation.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (TheBonheurs) [inline fooaccountInformation.json]: remove these 2 files

File: app/src/main/java/com/nervousfish/nervousfish/activities/ContactReceivedHelper.java [hunk 4]
```diff
@@ -109,6 +119,8 @@ private static void handleDuplicateContact(final Activity activity, final Contac
     }
 
     private static boolean checkKeysEqual(final List<IKey> keys1, final List<IKey> keys2) {
+        Validate.notNull(keys1);
+        Validate.notNull(keys2);
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/ContactReceivedHelper.java]: Use `assert` in private methods

File: app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java [hunk 5]
```diff
@@ -130,15 +138,16 @@ private void updateListPreference(final Preference preference, final String stri
      * immediately updated upon calling this method. The exact display format is
      * dependent on the type of preference.
      *
-     * @see #sBindPreferenceSummaryToValueListener
+     * @see #BIND_PREFERENCE_SUMMARY_TO_VALUE_LISTENER
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java]: Why not `{@see BIND_PREFERENCE_SUMMARY_TO_VALUE_LISTENER}`?

File: app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/Ed25519.java [hunk 6]
```diff
@@ -28,6 +30,8 @@
      * @param privateKey The generated private key.
      */
     private Ed25519(final EdDSAPublicKey publicKey, final EdDSAPrivateKey privateKey) {
+        Validate.notNull(publicKey);
+        Validate.notNull(privateKey);
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/Ed25519.java]: Use `assert` in private constructors

File: app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/EncryptorAdapter.java [hunk 7]
```diff
@@ -21,6 +22,7 @@
     // We suppress UnusedFormalParameter because the chance is big that a service locator will be used in the future
     @SuppressWarnings("PMD.UnusedFormalParameter")
     private EncryptorAdapter(final IServiceLocator serviceLocator) {
+        Validate.notNull(serviceLocator);
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/EncryptorAdapter.java]: Use `assert` in private constructors

File: app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java (hunk 5 above)
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java]: If you press Ctrl-Q, Intellij shows you the javadoc of the method. The @see is omitted if you wrap it in braces and we do not want that of course

Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java]: 🤔 and if you use `@link`?

Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java]: But @see works excellent! It's designed for this exact purpose. Why would I want to use something else?

Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java]: Ok 😅 my bad, I thought you we're trying to accomplish something completely different

**Pull request merged**